### DISCUSSION TITLE
Combine HDR10 and Dolby Vision

### DISCUSSION THREAD

**Discussion by janzen01** [quietvoid/dovi_tool#222] [Q&A]
Hello, I have two video files when one is HDR10 only, and the second is Dolby Vision only (on a non-DV display it has the wrong colors/brightness). I would like to combine these files into one, so on a non-DV display the HDR10 will be played and on DV display the Dolby Vision will be played. According to the HDR format provided by MediaInfo, it could be Dolby Vision, Version 1.0, dvhe.08.06, BL+RPU, Blu-ray compatible / SMPTE ST 
2086, HDR10 compatible

**Comment by hyllm** [comment]
convert RPU from profile 5 to profile 8.1 and then mux to hdr10 stream. :) also, you must check the sync. :)

**Reply by janzen01** [reply]
Files are the same length. Thank you for advice, I will try :)

**Reply by hyllm** [reply]
lenght is not all... you must check L5 data in DoVi... must match the HDR10 stream.

**Reply by janzen01** [reply]
Just to make sure I have the right procedure.
1. Extract HEVC from DV MKV
2. Demux DV-HEVC with conversion to 8.1 --> BL.HEVC & EL.HEVC
3. Extract HEVC from HDR MKV --> HDR.HEVC
4. Mux HDR.HEVC as BL and EL.HEVC as EL.

Do I understand this correctly?
Because when I try this, I am still getting SMPTE ST 2086, HDR10 compatible as output.
I'm a complete newbie :)

**Reply by Vippkeyz** [reply]
Hi, do you have succed ? Because i don't get it how to do ? And what adjustement i need to do for not having sync problem

**Reply by Shao85** [reply]
> Do I understand this correctly? Because when I try this, I am still getting SMPTE ST 2086, HDR10 compatible as output. I'm a complete newbie :)

You should probably end up with something like this: _**Dolby Vision, Version 1.0, dvhe.08.03, BL+RPU, HDR10 compatible / SMPTE ST 2086, HDR10 compatible**_
If the original "DV file" was encoded just about 2Mbps or below, then it is 10bit **MEL** (minimal enhancement layer) which works as storage for RPU data.
**FEL**(full enhancement layer) would be much bigger as it contains 16-bit TIFFs from which is derived the 12bit enhancement. You can see the difference between MEL and FEL after you mux them into container and play them. MEL will shows you nothing than green color. FEL will shows you delta from the TIFFs and BL (kind of dull grayish look)

**Reply by janzen01** [reply]
> Hi, do you have succed ? Because i don't get it how to do ? And what adjustement i need to do for not having sync problem

If I tried dovi_tool only, I failed. But I found a top set of tools (DDVT Tools) where it works. This set of tools is based on this dovi_tool project 🤩, but with a user-friendly interface. I hope I will not break any rules if I post the link to this set of tools here: http://forum.doom9.net/showthread.php?t=183479.
